File: 3347-distribute-elements-into-two-arrays-i/distribute-elements-into-two-arrays-i.py
class Solution:
    def resultArray(self, nums: List[int]) -> List[int]:
        arr1,arr2 = [nums[0]],[nums[1]]
        i = 2
        n = len(nums)
        while i<n:
            if arr1[-1]>arr2[-1]: arr1.append(nums[i])
            else: arr2.append(nums[i])
            i+=1
        return arr1+arr2

File: 3347-distribute-elements-into-two-arrays-i/distribute-elements-into-two-arrays-i.cpp
class Solution {
public:
    vector<int> resultArray(vector<int>& nums) {
        vector<int> arr1 = {nums[0]};
        vector<int> arr2 = {nums[1]};
        int l1 = nums[0], l2 = nums[1];
        int i = 2;
        while (i<nums.size()){
            if (l1>l2){l1 = nums[i]; arr1.push_back(nums[i++]);}
            else {l2 = nums[i]; arr2.push_back(nums[i++]);}
        }
        arr1.insert(arr1.end(),arr2.begin(),arr2.end());
        return arr1;
    }
};